packages/server-wallet/src/wallet/store.ts

```diff
@@ -697,6 +699,13 @@ async function createChannel(
   fundingLedgerChannelId: Bytes32 | undefined,
   txOrKnex: TransactionOrKnex
 ): Promise<Channel> {
+  if (fundingLedgerChannelId) {
+    const ledger = await Channel.forId(fundingLedgerChannelId, txOrKnex);
+    if (!ledger) throw new StoreError(StoreError.reasons.invalidFundingLedgerChannelId);
+    if (ledger.supported && _.some(ledger.support, 'isFinal'))
```

Comment:
Not sure I understand this. Is the channel `closing` as soon as I have received an `isFinal` state? Does that state need to be in `ledger.support`? Or should we check against `ledger.vars` ?

Reply:
Should we have a `.status()` getter on the `Channel` model, so that we don't have to keep revisiting channel statuses and recomputing them ad hoc?

Reply:
Or we use a `ledger.isRunning` method?

Reply:
This is not in the protocol code, just in the `createChannel` method on the `Store`. In theory it's possible you might be closing the ledger channel (proposed an `isFinal` state) and then later decide that you don't want to close it, but this logic ignores that. The logic is "throw an error if trying to create a channel funded by a ledger channel with any `isFinal` state in the support). The user could always call `createChannel` again later. For now, this is enough to prevent an issue and is at worst _too strict_ versus _too liberal_

Reply:
I think it might be too liberal, still. 

If I have proposed an `isFinal` state that is currently unsupported, my counterparty can unilaterally `conclude` the channel with that state at any point in the future. Deciding later to not close it is not an option, I think -- so the state should instantly and irrecoverably become `!running`.

 So I think it is unsafe to fund any new application channels with a ledger channel once I have signed an `isFInal` state.